metrics: Tersely show the commits in the size report.

jepler opened the pull request:
### Summary

In the size report, show information about the compared refs:
```
Reference:  00a926e99e zephyr/boards: Add PocketBeagle 2…
Comparison: 71ba57acca metrics: Tersely show the commits…
```

This will be helpful when checking which revision of the code size report comment on a PR corresponds to which revision of the code.

### Testing

I ran the ci.sh steps locally and inspected the output.

jepler committed:
metrics: Tersely show the commits in the size report.

This will show a line for both the reference and comparison, e.g.,
```
Reference:  00a926e99e zephyr/boards: Add PocketBeagle 2 rev A1…
Comparison: merge of c7ac411e22 metrics: Tersely show the commi…
```

When the comparison is a merge commit (as it is during CI) the second
parent of that commit is shown instead.

This will be helpful when checking which revision of the code size
report comment on a PR corresponds to which revision of the code.

Signed-off-by: Jeff Epler <jepler@unpythonic.net>

dpgeorge commented:
> This will be helpful when checking which revision of the code size report comment on a PR corresponds to which revision of the code.

Very nice idea!

> It's somewhat less useful than I hoped due to the comparison always being a merge ref...

Yeah that is a shame...

Maybe it's possible to rebase the PR to get a linear history??

jepler committed:
ci: Better code_size_build default reference ref.

Now the default reference commit is the first parent of the
selected commit, instead of the first parent of HEAD.

Signed-off-by: Jeff Epler <jepler@unpythonic.net>

jepler committed:
ci: Implement Jos Verlinde's suggestion.

[I suggest squashing this]

Signed-off-by: Jeff Epler <jepler@unpythonic.net>

dpgeorge commented:
Merged in 04323ad708d45720cfa6bc08ec7bbfdb9279df11 and 6f4fd6aed9df777a53e765ca926a7a6e58d79b01
